Moved the action arrays outside of the loop.

jayRyan24 opened the pull request:
This way they won't be made anew each iteration. I then moved them to the init and made them instance based as they are used each time through, so they may as well be made once only.

This shouldn't affect anything, right? @dr-leustyin

jayRyan24 commented:
As a side question, why can't the matches just be all of them, and always pass them?

dr-leustyin commented:
What do you mean, 'always pass them'?

On Sat, Jun 9, 2018, 2:26 PM Jay Ryan <notifications@github.com> wrote:

> As a side question, why can't the matches just be all of them, and always
> pass them?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/2739#issuecomment-395989478>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Al_G9mKOodvi-cIFiRMdkMuDtDntKWzjks5t7BNqgaJpZM4UhbdF>
> .
>

jayRyan24 commented:
Why do we only pass them to the bput method based on if we need them? Why not just send the whole shabang at once? Just make them the whole set and pass them all?

jayRyan24 commented:
```
@matches = [/^You /, "You get", "You draw out", "You pick up", "You get","You draw out", "You pick up", "Roundtime", "You put", "You sheathe"]```
```
actions = game_state.next_aim_action.split(";")
actions.each do | action |
  bput(action.strip, @matches)
  pause 0.25
  waitrt?
end
```

rpherbig commented:
@jayRyan24 over in #2733, @dr-leustyin said
> The default /^You /, is better, but incomplete as it would sometimes be triggered by 'You are fully rested' or 'You block with your shield' messaging, so I added some more specific checks for a non-exhaustive set of potential actions.

Which implies to me that we only _sometimes_ want to use `/^You /`, and use the others when possible.

jayRyan24 commented:
Ah yes, thanks.

rpherbig commented on combat-trainer.lic:
```diff
@@ -2088,6 +2088,15 @@ class AttackProcess
     echo("  @ambush_location: #{@ambush_location}") if $debug_mode_ct
 
+    @get_actions = %w[get wield]
+    echo("  @get_actions: #{@get_actions}") if $debug_mode_ct
```
💡 I'm not sure if there's much value in echoing something that isn't variable. On the other hand, it's not that much extra logging.

jayRyan24 replied:
Thought this too, but it could be helpful with debugging what is expected vs what we put in. It's only output once, so I figured it would be ok.

rpherbig commented:
I'm fine with merging this @jayRyan24 - or did you want to wait on any further discussion with @dr-leustyin ?

jayRyan24 commented:
It's practically the same thing, just we don't constantly make new arrays.

jayRyan24 commented:
At a minimum, this could be done.
```
    elsif game_state.loaded
      actions = game_state.next_aim_action.split(";")
      get_actions = %w[get wield]
      rt_actions = %w[gouge attack jab feint draw lunge slice]
      stow_actions = %w[stow sheath put]
      actions.each do | action |
        first_word = action.split(" ")[0].downcase
        matches = [/^You /]
        matches = ["You get","You draw out","You pick up"] if get_actions.include?(first_word)
        matches = ["Roundtime"] if rt_actions.include?(first_word)
        matches = ["You put", "You sheathe"] if stow_actions.include?(first_word)
        bput(action.strip, matches)
        pause 0.25
        waitrt?
      end
```

jayRyan24 commented:
@dr-leustyin

dr-leustyin commented:
I've got no problems with the updates

On Sat, Jun 9, 2018, 6:14 PM Jay Ryan <notifications@github.com> wrote:

> @dr-leustyin <https://github.com/dr-leustyin>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/2739#issuecomment-396002576>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Al_G9gGLaqPMutlw6Di9i5bTzlN8qB45ks5t7EjigaJpZM4UhbdF>
> .
>

jayRyan24 commented:
@rpherbig Think we're good.
